src/Platform/PlatformEnums.h

```diff
@@ -107,104 +107,104 @@ using enum PlatformLoggerOption;
-    RightBracket,
-    Decimal,
-    Semicolon,
-    Period,
-    Slash,
-    SingleQuote,
-    BackSlash,
-    Tilde,
-
-    Left,
-    Right,
-    Up,
-    Down,
-
-    PrintScreen,
-
-    Insert,
-    Home,
-    End,
-    PageUp,
-    PageDown,
-    Delete,
-    Select,
-
-    Control,
-    Alt,
-    Shift,
-
-    Numpad0, Numpad1, Numpad2, Numpad3, Numpad4, Numpad5, Numpad6, Numpad7, Numpad8, Numpad9,
-
-    Gamepad_A,
-    Gamepad_B,
-    Gamepad_X,
-    Gamepad_Y,
-    Gamepad_Left,
-    Gamepad_Right,
-    Gamepad_Up,
-    Gamepad_Down,
-    Gamepad_L1,
-    Gamepad_R1,
-    Gamepad_L3,
-    Gamepad_R3 ,
-    Gamepad_Start,
-    Gamepad_Back,
-    Gamepad_Guide,
-    Gamepad_Touchpad,
-
-    Gamepad_LeftStick_Left,
-    Gamepad_LeftStick_Right,
-    Gamepad_LeftStick_Up,
-    Gamepad_LeftStick_Down,
-    Gamepad_RightStick_Left,
-    Gamepad_RightStick_Right,
-    Gamepad_RightStick_Up,
-    Gamepad_RightStick_Down,
-    Gamepad_L2,
-    Gamepad_R2,
-
-    None,
-
-    First = Char,
-    Last = Gamepad_R2,
+    KEY_F1,
+    KEY_F2,
+    KEY_F3,
+    KEY_F4,
+    KEY_F5,
+    KEY_F6,
+    KEY_F7,
+    KEY_F8,
+    KEY_F9,
+    KEY_F10,
+    KEY_F11,
+    KEY_F12,
+
+    KEY_DIGIT_0, KEY_DIGIT_1, KEY_DIGIT_2, KEY_DIGIT_3, KEY_DIGIT_4, KEY_DIGIT_5, KEY_DIGIT_6, KEY_DIGIT_7, KEY_DIGIT_8, KEY_DIGIT_9,
+
+    KEY_A, KEY_B, KEY_C, KEY_D, KEY_E, KEY_F, KEY_G,
+    KEY_H, KEY_I, KEY_J, KEY_K, KEY_L, KEY_M, KEY_N, KEY_O, KEY_P,
+    KEY_Q, KEY_R, KEY_S,
+    KEY_T, KEY_U, KEY_V,
+    KEY_W, KEY_X, KEY_Y, KEY_Z,
+
+    KEY_RETURN,
+    KEY_ESCAPE,
+    KEY_TAB,
+    KEY_BACKSPACE,
+    KEY_SPACE,
+
+    KEY_ADD,
+    KEY_SUBTRACT,
+    KEY_COMMA,
+    KEY_LEFTBRACKET,
+    KEY_RIGHTBRACKET,
+    KEY_DECIMAL,
+    KEY_SEMICOLON,
+    KEY_PERIOD,
+    KEY_SLASH,
+    KEY_SINGLEQUOTE,
+    KEY_BACKSLASH,
+    KEY_TILDE,
+
+    KEY_LEFT,
+    KEY_RIGHT,
+    KEY_UP,
+    KEY_DOWN,
+
+    KEY_PRINTSCREEN,
+
+    KEY_INSERT,
+    KEY_HOME,
+    KEY_END,
+    KEY_PAGEUP,
+    KEY_PAGEDOWN,
+    KEY_DELETE,
+    KEY_SELECT,
+
+    KEY_CONTROL,
+    KEY_ALT,
+    KEY_SHIFT,
+
+    KEY_NUMPAD_0, KEY_NUMPAD_1, KEY_NUMPAD_2, KEY_NUMPAD_3, KEY_NUMPAD_4, KEY_NUMPAD_5, KEY_NUMPAD_6, KEY_NUMPAD_7, KEY_NUMPAD_8, KEY_NUMPAD_9,
+
+    KEY_GAMEPAD_A,
+    KEY_GAMEPAD_B,
+    KEY_GAMEPAD_X,
+    KEY_GAMEPAD_Y,
+    KEY_GAMEPAD_LEFT,
+    KEY_GAMEPAD_RIGHT,
+    KEY_GAMEPAD_UP,
+    KEY_GAMEPAD_DOWN,
+    KEY_GAMEPAD_L1,
+    KEY_GAMEPAD_R1,
+    KEY_GAMEPAD_L3,
+    KEY_GAMEPAD_R3 ,
+    KEY_GAMEPAD_START,
+    KEY_GAMEPAD_BACK,
+    KEY_GAMEPAD_GUIDE,
+    KEY_GAMEPAD_TOUCHPAD,
+
+    KEY_GAMEPAD_LEFTSTICK_LEFT,
+    KEY_GAMEPAD_LEFTSTICK_RIGHT,
+    KEY_GAMEPAD_LEFTSTICK_UP,
+    KEY_GAMEPAD_LEFTSTICK_DOWN,
+    KEY_GAMEPAD_RIGHTSTICK_LEFT,
+    KEY_GAMEPAD_RIGHTSTICK_RIGHT,
+    KEY_GAMEPAD_RIGHTSTICK_UP,
+    KEY_GAMEPAD_RIGHTSTICK_DOWN,
+    KEY_GAMEPAD_L2,
+    KEY_GAMEPAD_R2,
+
+    KEY_NONE,
+
+    KEY_FIRST = KEY_CHAR,
+    KEY_LAST = KEY_GAMEPAD_R2,
```

Comment:
I believe we now use `FIRST_KEY` and `LAST_KEY` style.

Reply:
We usually name enums in the form `PREFIX_SMTH`, and so for `FIRST` / `LAST` it becomes `PREFIX_FIRST` / `PREFIX_LAST`. So `KEY_FIRST` is OK.